Null replaceValue and multiLine findRegex processing.

B-ru opened the pull request:
Removed "required" attribute of "replaceValue" parameter.

Empty String replacement for null "replaceValue" parameter added at ProcessFilesTask.processFileContents.

ProcessFilesTask.processFileContents method was modified to achieve multiline findRegex processing.

Corresponding tests were added:
* FindAndReplaceMojoTest.testFileContentsMultiLineRegex
* FindAndReplaceMojoTest.testFileContentsMultiLineRegex2

Removed redundant toUri() calls at several tests.

B-ru committed:
Null replaceValue and multiLine findRegex processing.

Removed "required" attribute of "replaceValue" parameter.

Empty String replacement for null "replaceValue" parameter added at ProcessFilesTask.processFileContents.

ProcessFilesTask.processFileContents method was modified to achieve multiline findRegex processing.

Corresponding tests were added:
* FindAndReplaceMojoTest.testFileContentsMultiLineRegex
* FindAndReplaceMojoTest.testFileContentsMultiLineRegex2

Removed redundant toUri() calls at several tests.

B-ru committed:
Merge remote-tracking branch 'floverfelt/main'

# Conflicts:
#	src/main/java/io/github/floverfelt/find/and/replace/maven/plugin/tasks/ProcessFilesTask.java

B-ru committed:
New test for findValue with line-separator.
Test method regexFileContains added.

B-ru committed:
Unnecessary statement deleted.

bherw commented:
I find this change very useful. However, I encountered an issue with it: when performing a find-and-replace on executable files, this execution does not preserve the executable bit.

B-ru commented:
> I find this change very useful. However, I encountered an issue with it: when performing a find-and-replace on executable files, this execution does not preserve the executable bit.

@bherw  Thanks for your response. As far as I can tell, this behavior with the execution bit is also typical for the parent code, isn't it?
Fixed for POSIX-compatible OS.

B-ru committed:
Access bits saving attemp.

B-ru committed:
Windows-dependant regular expression were replaced with system-independant ones in tests.
The verification and application of access rights has been changed to POSIX-compatible, the previous approach was erroneous. The verification and application of access attributes is omitted for non-posix compliant operating systems. The relevant tests have been implemented.
